src/rp2_common/pico_cyw43_arch/cyw43_arch_freertos.c

```diff
@@ -4,250 +4,66 @@
  * SPDX-License-Identifier: BSD-3-Clause
  */
 
-#include <stdio.h>
+#if PICO_CYW43_ARCH_FREERTOS
 
 #include "pico/cyw43_arch.h"
-
-#include "hardware/gpio.h"
-#include "hardware/irq.h"
-#include "hardware/sync.h"
-
-#include "cyw43_stats.h"
+#include "pico/cyw43_driver.h"
+#include "pico/async_context_freertos.h"
 
 #if CYW43_LWIP
+#include "pico/lwip_freertos.h"
 #include <lwip/tcpip.h>
 #endif
 
-#if PICO_CYW43_ARCH_FREERTOS
-
-// FreeRTOS includes
-#include "FreeRTOS.h"
-#include "timers.h"
-#include "semphr.h"
-
 #if NO_SYS
-#error example_cyw43_arch_frertos_sys requires NO_SYS=0
-#endif
-
-#ifndef CYW43_TASK_PRIORITY
-#define CYW43_TASK_PRIORITY ( tskIDLE_PRIORITY + 4)
+#error example_cyw43_arch_freetos_sys requires NO_SYS=0
 #endif
 
-#ifndef CYW43_SLEEP_CHECK_MS
-#define CYW43_SLEEP_CHECK_MS 50 // How often to run lwip callback
-#endif
-
-#define CYW43_GPIO_IRQ_HANDLER_PRIORITY 0x40
+static async_context_freertos_t cyw43_async_context_freertos;
 
-static void signal_cyw43_task(void);
-
-#if !LWIP_TCPIP_CORE_LOCKING_INPUT
-static SemaphoreHandle_t cyw43_mutex;
+async_context_t *cyw43_arch_init_default_async_context(void) {
+    async_context_freertos_config_t config = async_context_freertos_default_config();
+#ifdef CYW43_TASK_PRIORITY
+    config.task_priority = CYW43_TASK_PRIORITY;
 #endif
-static TimerHandle_t timer_handle;
-static TaskHandle_t cyw43_task_handle;
-static volatile bool cyw43_task_should_exit;
-static SemaphoreHandle_t cyw43_worker_ran_sem;
-static uint8_t cyw43_core_num;
-
-// Called in low priority pendsv interrupt only to do lwip processing and check cyw43 sleep
-static void periodic_worker(__unused TimerHandle_t handle)
-{
-#if CYW43_USE_STATS
-    static uint32_t counter;
-    if (counter++ % (30000 / LWIP_SYS_CHECK_MS) == 0) {
-        cyw43_dump_stats();
-    }
+#ifdef CYW43_TASK_STACK_SIZE
```

Comment:
Could /should we have a PICO_CONFIG entry for this? Similarly for CYW43_TASK_PRIORITY?

Reply:
yes; for some reason i thought you had another PR dealing with these

Reply:
I had this https://github.com/raspberrypi/pico-sdk/pull/1134. I just closed it though as I decided there was nothing more to do. I was debating increasing the default for the stack size but the ping and iperf examples seem to be ok with 1024 bytes.